packages/dd-trace/src/appsec/waf/waf_context_wrapper.js

```diff
@@ -26,7 +26,9 @@ class WAFContextWrapper {
 
     // TODO: possible optimizaion: only send params that haven't already been sent with same value to this wafContext
     for (const key of Object.keys(params)) {
-      if (this.requiredAddresses.has(key) && !this.addressesToSkip.has(key)) {
+      // TODO: requiredAddresses is no longer used due to processor addresses are not included in the list. Check on
```

Comment:
Why could we send more addresses than necessary instead of just adding `waf.context.processor` to the `requiredAddresses`?

Reply:
Yeah we could also hardcode it too if that's the only address yes. The real solution would be to implement what the comment says: get the addresses list from diagnostics

Reply:
I had a chat with Anil about this and he kind of say that we were abusing this property and it use is more about to get the known addresses by the WAF but it does not include scanner addresses. He also mentioned that in future releases he could add the scanners addresses in the diagnostics so we can use them the way we were using the regular ones.
Anyway it could happen the opposite in which we manually include addresses which the WAF is not aware of and we will be converting them anyway.

Reply:
Not sure I understand what you're saying here

Reply:
just saying that if you're removing all the feature and tests about `requiredAddresses` then you should do it correctly: 
![image](https://github.com/DataDog/dd-trace-js/assets/70517128/54fd0186-1282-41bb-8af7-2514f9a2f43b)

Reply:
Fixed